commit 668f612fa0d8d4120ec5dc0725d7e1ca3152a954
Author: Avi Kivity <avi@qumranet.com>
Date:   Wed Jul 2 09:28:55 2008 +0300

    KVM: VMX: Move nmi injection failure processing to vm exit path
    
    Instead of processing nmi injection failure in the vm entry path, move
    it to the vm exit path (vm_complete_interrupts()).  This separates nmi
    injection from nmi post-processing, and moves the nmi state from the VT
    state into vcpu state (new variable nmi_injected specifying an injection
    in progress).
    
    Signed-off-by: Avi Kivity <avi@qumranet.com>

diff --git a/include/asm-x86/kvm_host.h b/include/asm-x86/kvm_host.h
index 4f2bd884fd3a..7cf69fd1dcfe 100644
--- a/include/asm-x86/kvm_host.h
+++ b/include/asm-x86/kvm_host.h
@@ -300,6 +300,7 @@ struct kvm_vcpu_arch {
 	struct page *time_page;
 
 	bool nmi_pending;
+	bool nmi_injected;
 
 	u64 mtrr[0x100];
 };